R/data-importer-configuration.R

```diff
@@ -0,0 +1,356 @@
+#' @title DataImporterConfiguration
+#' @docType class
+#' @description Configuration of data import from excel or csv files. To be used with #TODO
+#' @export
+#' @format NULL
+DataImporterConfiguration <- R6::R6Class(
+  "DataSet",
+  inherit = DotNetWrapper,
+  cloneable = TRUE,
+  active = list(
+    #' @field timeColumn Name of the column for time values
+    timeColumn = function(value) {
+      column <- private$.timeColumn
+      if (missing(value)) {
+        return(rClr::clrGet(column, "ColumnName"))
+      }
+      validateIsString(value)
+      rClr::clrSet(column, "ColumnName", enc2utf8(value))
+    },
+
+    #' @field timeUnit If \code{timeUnitFromColumn} is \code{FALSE}, unit of the values in time column
+    #' If \code{timeUnitFromColumn} is \code{TRUE}, name of the column with units of the values in time column
+    timeUnit = function(value) {
```

Comment:
Time unit (the same is valid for measurement unit) - it is either a fixed unit that the user can change, or the name of the column that defines the units.